src/dune_package.ml

```diff
@@ -30,8 +30,18 @@ module Lib = struct
   let make ~loc ~kind ~name ~synopsis ~archives ~plugins ~foreign_objects
         ~foreign_archives ~jsoo_runtime ~main_module_name ~sub_systems
         ~requires ~ppx_runtime_deps ~implements ~virtual_ ~modules ~modes
-        ~version ~orig_src_dir ~dir =
-    let map_path p = Path.relative dir (Path.basename p) in
+        ~version ~orig_src_dir ~dir ~is_external =
+    let map_path =
+      if is_external then
+        fun p ->
+          match Path.kind p with
+          | External _ -> p
+          | Local _ ->
+            Exn.code_error "Lib.make got local path for an external library"
+              [ "path", Path.to_sexp p ]
```

Comment:
Can't we just omit the flag and use this branch for local libs?

Reply:
Yes, that makes sense

Reply:
Done